storage-proxy/nginx.conf

```diff
@@ -0,0 +1,96 @@
+worker_processes  auto;
+
+error_log  /dev/stdout warn;
+pid        /var/run/nginx.pid;
+
+events {
+    worker_connections  1024;
+}
+
+env DOWNLOAD_SPEED;
+env MAX_CONNECTIONS;
+
+http {
+
+    init_by_lua_block {
+        ngx.log(ngx.WARN, "download speed limit: " .. (os.getenv("DOWNLOAD_SPEED") or "not set"))
+        ngx.log(ngx.WARN, "max connections: " .. (os.getenv("MAX_CONNECTIONS") or "not set"))
+    }
+
+    upstream minio_backend {
+        server minio:9000 max_fails=0;
+    }
+
+    lua_shared_dict my_limit_conn_store 100m;
+
+    server {
+
+        proxy_max_temp_file_size 0;
+
+        listen 9000;
+
+        location / {
+            access_by_lua_block {
```

Comment:
`limit_rate` does not work inside `if` block (or at least I could not get past nginx complaining that limit_rate cannot be used in `if` configuration block). It wouldn't be possible to use `if ($request_method == GET) { ... }` and we obviously do not want to limit deployments -> minio upload. Given this, we'd have to use a separate prefix, what breaks request signing for all incoming requests (URL is part of data being signed).

Another alternative is allowing deployments to access minio directly and applying `limit_rate` on `location /` but again signing will be broken because host is part of signature as well.

Lua also makes environment access easy. An alternative is to generate a part of (or a whole) `nginx.conf` with `envsubst`, meaning we'd have to use a custom entrypoint to preform substitution, hence possibly an Dockerfile and another repository. But this does not resolve either of the first 2 problems.